mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryMailboxMapper.java

```diff
@@ -158,24 +158,24 @@ public void updateACL(Mailbox mailbox, MailboxACL.MailboxACLCommand mailboxACLCo
         mailbox.setACL(mailbox.getACL().apply(mailboxACLCommand));
     }
 
+    private synchronized Iterable<MailboxAnnotation> retrieveAllAnnotations(InMemoryId maiboxId) {
```

Comment:
what if using the Lock as current updated code? Let me know such as I've never used it before.